src/libsync/vfs/cfapi/hydrationjob.cpp

```diff
@@ -184,11 +271,36 @@ void OCC::HydrationJob::onNewConnection()
     Q_ASSERT(!_transferDataSocket);
     Q_ASSERT(!_job);
 
-    qCInfo(lcHydration) << "Got new connection starting GETFileJob" << _requestId << _folderPath;
-    _transferDataSocket = _transferDataServer->nextPendingConnection();
-    _job = new GETFileJob(_account, _remotePath + _folderPath, _transferDataSocket, {}, {}, 0, this);
-    connect(_job, &GETFileJob::finishedSignal, this, &HydrationJob::onGetFinished);
-    _job->start();
+    if (isEncryptedFile()) {
```

Comment:
maybe consider splitting this method into two methods onNewConnectionEncryptedFile and on NewConnectionNormalFile

Reply:
@FlexW I am not a fun of having 2 different signals and connecting one or another based on the condition. Having said that, I will move the code from its internal condition to 2 separate methods. Hopefully, this will make the code more readable.